assets/js/lib/test-utils/factories/clusters.js

```diff
@@ -112,7 +117,10 @@ export const clusterFactory = Factory.define(({ sequence, params }) => {
   };
 });
 
-export const addHostsToAscsErsClusterDetails = (details) =>
+export const buildHostsFromAscsErsClusterDetails = (details) =>
   details.sap_systems
     .flatMap(({ nodes }) => nodes)
     .map(({ name }) => hostFactory.build({ hostname: name }));
+
+export const buildSapSystemsFromAscsErsClusterDetails = (details) =>
+  details.sap_systems.map(({ sid }) => sapSystemFactory.build({ sid }));
```

Comment:
The only super nitpicky thing I can comment is that I'm not a huge fan of utility functions inside the factories file, as creating the right dataset is responsibility (in theory) of the test itself.

Maybe we could move this to the test directly? Your call :smile:

Reply:
The thing is that these functions are not only used in the tests. They are needed for storybook as well.
In fact, I consider them part of the factory, as they complement the data creation.
We have multiple utilities like this in other factories, specially in the `execution.js` with the `with...` syntax.

In any case, if I move them to the tests, I would need to import them to storybook files from the tests, or duplicate the function.
What is the best solution for you? I would vote for keeping as it is, but I want to know your opinion hehe

Reply:
Ouch, you are right. Then I think this is the most viable tradeoff. LGTM :smile: